src/components/clusters/ClusterStatus.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Popover, Button, ButtonVariant } from '@patternfly/react-core';
+import {
+  global_danger_color_100 as dangerColor,
+  global_warning_color_100 as warningColor,
+  global_success_color_200 as okColor,
+} from '@patternfly/react-tokens';
+import {
+  ExclamationCircleIcon,
+  WarningTriangleIcon,
+  CheckCircleIcon,
+  UnknownIcon,
+  InProgressIcon,
+} from '@patternfly/react-icons';
+import { Cluster } from '../../api/types';
+import { CLUSTER_STATUS_LABELS } from '../../config/constants';
+
+const getStatusIcon = (status: Cluster['status']) => {
+  if (status === 'insufficient') return <WarningTriangleIcon color={warningColor.value} />;
```

Comment:
What about `switch` without `default` so linter can warn us to add a section when the type changes?